beacon/validator/block_builder.go

```diff
@@ -41,21 +41,19 @@ import (
 
 // BuildBlockAndSidecars builds a new beacon block.
 //
-//nolint:funlen
+
 func (s *Service) BuildBlockAndSidecars(
 	ctx context.Context,
 	slotData *types.SlotData,
 ) ([]byte, []byte, error) {
-	var (
-		signedBlk *ctypes.SignedBeaconBlock
-		blk       *ctypes.BeaconBlock
-		sidecars  datypes.BlobSidecars
-		forkData  *ctypes.ForkData
-	)
-
 	startTime := time.Now()
 	defer s.metrics.measureRequestBlockForProposalTime(startTime)
 
+	if !s.localPayloadBuilder.Enabled() {
```

Comment:
a minor optimization: if block building is disabled (as it should be, e.g. in full nodes) we could return early here

Reply:
Q: do we know if PrepareProposal is ever called for a non-validator node?